### PR TITLE
fix: importFrom to expose custom-media for use

### PR DESCRIPTION
**Pull request by zchsh**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-zsimport-postcss-custom-media-hashicorp.vercel.app) 🔎
- [Asana task](https://app.asana.com/0/1201987349274776/1202080355906373/f) 🎟️

## What

Adds our `custom-media` queries specific to `dev-portal` to our `postcss.config.js`.

## Why

So that we can more reliably use `postcss-custom-media`. For example, `@media (--dev-dot-tablet-up) {}`.

## How

Clones and modifies the imported config from `@hashicorp/platform-postcss-config`

## Testing

- [ ] Visit a tutorial page, such as [`/vault/tutorials/getting-started-ui/getting-started-intro-ui`][tutorial-view], and verify that `TutorialMeta`'s use of `custom-media` still works
    - On viewports larger than 1000px, you should see some media-query styles applied. These indicate that `custom-media` is working as expected
- [ ] Visit an install page, such as [`/waypoint/downloads`][install-view], and verify that `custom-media` on `.cardGrid` works as expected
    - Note: specific `@media` styles will only be visible on viewports less than `1000px`. At viewports wider than `1000px`, there's no expectation to see `@media` styles.

## Anything else?

Nothing urgent. We may later want to support this in `@hashicorp/platform-postcss-config`, but for now this is only use case we have for extending `importFrom`, so we're choosing not to build that abstraction at the moment.

[tutorial-view]: https://dev-portal-git-zsimport-postcss-custom-media-hashicorp.vercel.app/vault/tutorials/getting-started-ui/getting-started-intro-ui
[install-view]: https://dev-portal-git-zsimport-postcss-custom-media-hashicorp.vercel.app/waypoint/downloads

### CODE REVIEW

File: config/redirects.js [hunk 1]
```diff
@@ -253,7 +253,7 @@ async function buildDotIoRedirects() {
     : await fetchGithubFile({
         owner: 'hashicorp',
         repo: 'consul',
-        path: 'website/redirects.next.js',
```
Comment (zchsh) [inline config/redirects.js:256]: This is an unrelated fix, builds fail without it. See https://github.com/hashicorp/consul/pull/12696 which renamed the file.

File: src/pages/style.css [hunk 2]
```diff
@@ -115,18 +115,8 @@
...
 /*
-***
-* DevDot
-***
-* The custom media and classes below are helpers used in DevDot.
-***
+Mobile & tablet visibility helpers, specific to dev-dot.
 */
-
-@custom-media --dev-dot-mobile only screen and (max-width: 728px);
```
Comment (zchsh) [inline src/pages/style.css:125]: Moved these to a dedicated `custom-media` file, which we now `importFrom` in `postcss.config.js`. I think this will make these more reliably available. (Honestly not exactly sure how they're available as-is).

File: src/views/product-downloads-view/components/featured-tutorials-section/featured-tutorials-section.module.css [hunk 3]
```diff
@@ -16,11 +16,11 @@ out.
   grid-column-gap: 24px;
   grid-row-gap: 24px;
 
-  @media only screen and (min-width: 729px) and (max-width: 1000px) {
+  @media (--dev-dot-tablet) {
```
Comment (zchsh) [inline src/views/product-downloads-view/components/featured-tutorials-section/featured-tutorials-section.module.css:19]: Made these changes to demo that this change _seems_ to work. Also checked the `/tutorials` view, it seems that `TutorialsMeta` retains the custom-media queries it was using (which were already working).

File: config/redirects.js (hunk 1 above)
Comment (BRKalow) [reply config/redirects.js:256]: Fixed in https://github.com/hashicorp/dev-portal/commit/010485344365b9c474dcb687975e05d0186dcdb7

Comment (zchsh) [reply config/redirects.js:256]: ah nice, i'll rebase and drop this 💖

File: postcss.config.js [hunk 4]
```diff
@@ -1,3 +1,42 @@
...
+
+/**
+ * TODO: this is a spike to show that this configuration may resolve
+ * the issues we're having in dev-dot when using @custom-media.
+ * We likely want to incorporate this into `@hashicorp/platform-postcss-config`.
+ * Eg, maybe it should accept an optional "options" object, where we can
+ * pass additional files to `importFrom`?
+ */
+function alsoImportDevDotCustomMedia(postcssConfig) {
+  const newPlugins = postcssConfig.plugins.map((p) => {
```
Comment (BRKalow) [inline postcss.config.js:11]: nit, naming for clarity:
```suggestion
  const newPlugins = postcssConfig.plugins.map((plugin) => {
```

File: postcss.config.js [hunk 5]
```diff
@@ -1,3 +1,42 @@
...
+    // we only want to modify postcss-preset-env
+    const isPresetEnv =
+      Array.isArray(p) && p.length == 2 && p[0] == 'postcss-preset-env'
+    if (!isPresetEnv) return p
+    // we want to modify the postcss-preset-env settings object,
```
Comment (BRKalow) [inline postcss.config.js:16]: We could destructure this to make the following code a bit easier to grok, instead of using `p[1]`.

```ts
const [, presetEnvOptions] = plugin
```

File: postcss.config.js [hunk 6]
```diff
@@ -1,3 +1,42 @@
...
+    const newPresetEnvSettings = {
+      ...p[1],
+      features: {
+        ...p[1].features,
+        'custom-media-queries': {
+          ...p[1].features['custom-media-queries'],
+          importFrom: newImportFrom,
+        },
+      },
+    }
```
Comment (BRKalow) [inline postcss.config.js:33]: total preference, but I think mutating this would be okay and maybe clearer:

```ts
const newPresetEnvSettings = [...presentEnvOptions]
newPresetEnvSettings.features['custom-media-queries'].importFrom = newImportFrom
```

**Review by BRKalow — APPROVED** [review]
Nothing blocking! Left a few minor comments, feel free to address or ignore

File: postcss.config.js (hunk 6 above)
Comment (zchsh) [reply postcss.config.js]: ✅ I dig it - updated!

File: postcss.config.js (hunk 4 above)
Comment (zchsh) [reply postcss.config.js]: ✅ Have updated this throughout the function body - I dig the clarity 👍

File: postcss.config.js (hunk 5 above)
Comment (zchsh) [reply postcss.config.js:18]: ✅ Updated!

File: postcss.config.js [hunk 7]
```diff
@@ -1,3 +1,39 @@
...
+ * pass additional files to `importFrom`?
+ */
+function alsoImportDevDotCustomMedia(postcssConfig) {
+  const newPlugins = postcssConfig.plugins.map((plugin) => {
+    // we only want to modify postcss-preset-env
+    const isPresetEnv =
+      Array.isArray(plugin) &&
+      plugin.length == 2 &&
+      plugin[0] == 'postcss-preset-env'
+    if (!isPresetEnv) return plugin
```
Comment (ashleemboyer) [inline postcss.config.js:17]: Does ESLint not run on this file? I would expect an error to happen here for the `curly` rule. CC @BRKalow

Comment (BRKalow) [reply postcss.config.js:17]: Weird, yeah, it looks like this file isn't getting picked up.
